### PR TITLE
Added some MvcOptionsExtensions.

### PR DESCRIPTION
**Pull request by PureKrome**
- WithGlobalAuthorization
- WithGlobalCancelledRequestHandler

### CODE REVIEW

**Comment by PureKrome** [comment]
Fixes #3

File: src/Homely.AspNetCore.Mvc.Helpers/Extensions/MvcOptionsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿using Homely.AspNetCore.Mvc.Helpers.Filters;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Authorization;
+
+namespace Homely.AspNetCore.Mvc.Helpers.Extensions
```
Comment (RPM1984) [inline src/Homely.AspNetCore.Mvc.Helpers/Extensions/MvcOptionsExtensions.cs:6]: Note: need to test this out to see how it works with our current JSON exception handling.

**Review by RPM1984 — APPROVED** [review]
Looks good... can we look at consolidating the two types of error handling (`app.UseExceptionHandler()` vs `ExceptionFilter) into one place? Maybe having the types of exceptions we want to handle each in their own filter, which we can optionally build up one by one?

**Comment by PureKrome** [comment]
👀 @RPM1984 

I've added tests for the 2x new (optional) global extension methods.

Also, some code cleanup.

Finally, I noticed that we might be consistent with the ApiError model across _all_ scenario's. So I'll make another card and we can chat about it, later.

**Comment by PureKrome** [comment]
NOTE: After this PR, I think this (again) needs to be reviewed with ASPNET-Core 2.1 release stuff.

**Pull request closed without merging**